src/js/osweb/classes/syntax.js

```diff
@@ -222,6 +222,39 @@ export default class Syntax {
     // return line.replace(/(?<!\\)\\(?=['"\\])/mg, '')
     return line.replace(/\\(?=['"])/mg, '').replace(/\\\\/mg, '\\')
   }
+  
+  /**
+   * Removes tab indentation from a script, if all lines are indented by a
+   * single tab.
+   * @param {String} script - A script
+   * @return {String} - A dedented script
+   */
+  dedent (script) {
+    const lines = script.split('\n')
+    let dedented = []
+    for (const line of lines) {
+      if (line[0] !== '\t') {
+        return script

```

Comment:
This seems peculiar. If the first character of a line is not a tab, return the script (e.g. the input variable?) so for instance a script like
```
    define something
    do something else
non-indented line
```
would just return the script since the last line does not start with a tab, even though you might still want to dedent the first two lines?
I assume this is just my lack of understanding the situation and you have tested this (judging from the unit tests) so it just works.

Reply:
Yes, that's intended. Basically, it should dedent a script that corresponds to an item definition, and this is always indented by a single tab. If any line is not indented, then it's not an item definition, and it should be left as is.